### ISSUE TITLE
Route on Map

### ISSUE THREAD

**Issue by mersay** [airbnb/react-native-maps#778]
is there any plans to make routes between the two markers show on the map??

**Comment by symmetriccurve** [comment]
You can simply use any Set of polyline data to route between two markers just have markers on first and last coordinates of polyline data

**Comment by yellowbang** [comment]
<img width="307" alt="screen shot 2016-11-21 at 7 20 55" src="https://cloud.githubusercontent.com/assets/6099830/20509740/ad3c0a9a-b01f-11e6-8e4a-134154385477.png">
This is what I have. When a street is not straight, it will has problem.
Any suggestions?

**Comment by symmetriccurve** [comment]
Get the polyline data from google routes like services so that you have polyline running through the roads.

**Comment by yellowbang** [comment]
Yes, that is what i did. However the polyline data from google routes are the coordinates of the starting and ending point of a street. If a street is straight, the polyline fits it well, but when a street is curved (eg, the Market St in my last screenshot), polyline is not working well as expect. 

by the way, this is the url I use for getting the route data:
'https://maps.googleapis.com/maps/api/directions/json?origin=' + startPosition + '&destination=' + endPosition + '&departure_time='+Date.now()+'&traffic_model=best_guess&key=' + Util.GOOGLE_API_KEY;
   
Thanks.

**Comment by mCzolko** [comment]
You should use the DirectionsService from Google API (If it is available on SDK on mobiles I know that on JS is). When you ask for routes via URL (as you did) you will get up to 50 major points coordinates. Could you imagine big will be the response JSON if you asked for directions from New York to SF? Too big. Please close this issue. The issue is not within this library but in your business logic.

Please read more here: https://developers.google.com/maps/documentation/javascript/directions

I'm not sure if there is a support for Map SDK (but is should be)

**Comment by ericapply** [comment]
@yellowbang You need to use overview_polyline.points data
```
"overview_polyline": {
        "points": "--------"
      }
```
You can decode it using https://github.com/mapbox/polyline or something similar

**Comment by bramus** [comment]
Based upon the feedback here (and in https://github.com/airbnb/react-native-maps/issues/929) I've knocked up a standalone and immediately usable component to do just that: https://github.com/bramus/react-native-maps-directions

**Comment by flyandi** [comment]
I just released a new library that encompasses all of this.. https://github.com/flyandi/react-native-maps-navigation
